src/mm-input-mask/mm-input-mask.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function () {
+
+	_types = {
+		SEP:0,
+		GROUP:1
+	};
+
+	_restrict = {
+		alpha:/[A-Za-z]+/g,
+		numeric:/\d+/ig,
+		alphaNumeric:/[A-Za-z\d]+/g,
+		all:/.+/g
+	};
+
+	_applyReg = function(reg, val) {
+		if (reg) {
+			reg.lastIndex = 0;
+			return reg.test(val);
+		}
+		return false;
+	};
+
+	_rules = {
+		hour12:function(char, input, group, field) {
+
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+			if (first) {
+				if (imp > 1) {
+					field.value = "0" + input;
+					return false;
+				}
+			} else if (second) {
+				if (imp > 2 && parseInt(input) === 1) {
+					field.value = input + "2";
+					return false;
+				}
+			}
+			return true;
+		},
+		hour24: function(char, input, group, field) {
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+		},
+		minutes: function(char, input, group, field) {
```

Comment:
I guess my recommendation would be towards a solution that only (and always) overwrites the input value if the intended content can't be coerced into a suitable form -- probably by doing validation after the fact with a "changed-handler" (or something along those lines).